### PR TITLE
fix: [Session] `Redis` connect to protocol `TLS`

### LINKED ISSUE #7178: Bug: Can't connect to Redis with Session RedisHandler over TLS
### PHP Version

8.1

### CodeIgniter4 Version

4.3.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### What happened?

I use a MS Azure Redis server which needs to be connected over TLS. When i set the $sessionDriver to the RedisHandler in the App.php file i edited the savePath to: 
`tls://nameofserver.redis.cache.windows.net:6380?auth=password`

When i debug the '/framework/system/Session/Handlers/RedisHandler.php' and check the '$this->savePath'. i see the following output because of the preg_match():
 
`Array
(
    [host] => tls
    [port] => 6379
    [password] => 
    [database] => 0
    [timeout] => 0
)`

This results in the Redis server not connecting and it gives a RedisException error. 

### Steps to Reproduce

Add 'tls://' to the Redis database server url.

### Expected Output

`Array
(
    [host] => tls://nameofserver.redis.cache.windows.net
    [port] => 6380
    [password] => password
    [database] => 0
    [timeout] => 0
)`

### Anything else?

I managed to bypass this (hopefully temporary) by copying the system folder in our project folder and hardcoded the Redis Server URL.

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Fixed #7178
Ref https://3v4l.org/kM3JD

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by kenjis — APPROVED** [review]
LGTM!

**Review by michalsn — APPROVED** [review]

File: system/Session/Handlers/RedisHandler.php [hunk 1]
```diff
@@ -102,20 +103,21 @@ protected function setSavePath(): void
             throw SessionException::forEmptySavepath();
         }
 
-        if (preg_match('#(?:tcp://)?([^:?]+)(?:\:(\d+))?(\?.+)?#', $this->savePath, $matches)) {
-            if (! isset($matches[3])) {
-                $matches[3] = ''; // Just to avoid undefined index notices below
+        if (preg_match('#(?:(^.*)://)?([^:?]+)(?:\:(\d+))?(\?.+)?#', $this->savePath, $matches)) {
```
Comment (samsonasik) [inline system/Session/Handlers/RedisHandler.php:106]: probably better to use possible values, eg: `(tcp|tls)`, that's we can have less possible regression.

Comment (ddevsr) [reply system/Session/Handlers/RedisHandler.php]: Done

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @ddevsr

**Pull request merged**